contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -1687,3 +1695,112 @@ allocates resources such as `NodePorts` and `ClusterIPs` and automatically fill
 These resources and the fields representing them are automatically cleared when  the users changes 
 service type to `ExternalName` where these resources and field values no longer apply. 	
+
+## Representing Allocated Values
+
+Many API types include values that are allocated on behalf of the user from
+some larger space (e.g. IP addresses from a range, or storage bucket names).
+These allocations are usually driven by controllers asynchronously to the
+user's API operations.  Sometimes the user can request a specific value and a
+controller must confirm or reject that request.  There are many examples of
+this in Kubernetes, and there a handful of patterns used to represent it.
+
+The common theme among all of these is that the system should not trust users
+with such fields, and must verify or otherwise confirm such requests before
+using them.
+
+Some examples:
+
+* Service `clusterIP`: Users may request a specific IP in `spec` or will be
+  allocated one (in the same `spec` field).  If a specific IP is requested, the
+  apiserver will ensure that IP is available or else will reject the API
+  operation synchronously (rare).  Consumers read the result from `spec`.  This
+  is safe because the value is either valid or it is never stored.
+* Service `loadBalancerIP`: Users may request a specific IP in `spec` or will
+  be allocated one which is reported in `status`.  If a specific IP is
+  requested, the LB controller will ensure that IP is available or else will
+  report failure asynchronously.  Consumers read the result from `status`.
+  This is safe because most users do not have acces to write to `status`.
+* PersistentVolumeClaims: Users may request a specific PersistentVolume in
+  `spec` or will be allocated one (in the same `spec` field).  If a specific PV
+  is requested, the volume controller will ensure that the volume is available
+  or else will report failure asynchronously.  Consumers read the result by
+  examining both the PVC and the PV.  This is more complicated than the others
+  because the spec value is stored before being confirmed, which could lead to
+  a user accessing someone else's PV.
+
+A counter-example:
+
+* Service `externalIPs`: Users must specify one or more specific IPs in `spec`.
+  The system cannot easily verify those IPs (by their definition, they are
+  external). Consumers read the result from `spec`.  This is UNSAFE and has
+  caused problems with untrusted users.
+
+In the past, API conventions dictated that `status` fields always come from
+observation, which made some of these cases more complicated than necessary.
+The conventions have been updated to allow `status` to hold such allocated
+values.  This is not a one-size-fits-all solution, though.
+
+### When to use a `spec` field
+
+New APIs should almost never do this.  Instead, they should use `status`.
+PersistentVolumes might have been simpler if we had done this.
```

Comment:
It seems strange to have an API recommendation of "when to _X_" that says "never do _X_".

I think it's reasonable to suggest that controllers should always use `status` for this reporting, but maybe we don't need a "when to use a `spec` field" section given the historical context above?

```suggestion
New APIs should almost never use `spec` fields to report allocated values.  Instead, they should use `status`.
PersistentVolumes might have been simpler if we had done this.
```

Reply:
I did it this way because it answered an actual design pattern I saw in the wild - (ab)using spec. I thought it makes the point string by saying "When should I X?  Almost never."  I liked the cheekiness.

I'll change it if reviewers agree, but my inclination is to leave it